tests/harness/model/test.py

```diff
@@ -0,0 +1,207 @@
+                return result, message
+
+        return True, None
+
+    @contextmanager
+    def setup_test_stack(
+        self,
+        client: "Client",
+        environment: "TestEnvironment",
+        module: Optional[ModuleType] = None,
+        cleanup: bool = True,
+    ) -> Generator["Stack", None, None]:
+        """Provision and activate a ZenML stack for this test.
+
+        Args:
+            module: A pytest test module.
+            client: The ZenML client to be used to configure the ZenML stack.
+            environment: The environment providing the stack requirements.
+            cleanup: Whether to clean up the stack after the test.
+
+        Yields:
+            The active stack that the test should use.
+        """
+        from zenml.enums import StackComponentType
+        from zenml.utils.string_utils import random_str
+
+        components: Dict[StackComponentType, Union[str, UUID]] = {}
+
+        stack_requirements = [req.stacks for req in self.compiled_requirements]
+        requirements = itertools.chain.from_iterable(stack_requirements)
+
+        for stack_requirement in requirements:
+            component = stack_requirement.find_stack_component(
+                client, environment=environment
+            )
+            if component is None:
+                # This should not happen if the requirements are checked
+                # before calling this method.
+                raise RuntimeError(
+                    f"could not find a stack component that matches the test "
+                    f"requirements '{stack_requirement}'."
+                )
+            if component.type in components:
+                raise RuntimeError(
+                    f"multiple stack components of type '{component.type}' "
+                    f"are specified as requirements by the test requirements."
+                )
+            components[component.type] = component.id
+
+        # Every stack needs an orchestrator and an artifact store.
+        for component_type in [
+            StackComponentType.ORCHESTRATOR,
+            StackComponentType.ARTIFACT_STORE,
+        ]:
+            if component_type not in components:
+                requirement = StackRequirement(
+                    type=component_type,
+                )
+                component = requirement.find_stack_component(
+                    client, environment=environment
+                )
+
+                # There is always a default orchestrator and artifact store.
+                assert component is not None
+                components[component_type] = component.id
+
+        # Add remaining components enforced by the test environment.
+        for (
+            component_type,
+            mandatory_components,
+        ) in environment.mandatory_components.items():
+            if component_type not in components:
+                components[component_type] = mandatory_components[0].id
+
+        random_name = "pytest-"
```

Comment:
would we mayber want to also know at a glance which environment this stack is for?

Reply:
the stack names are already quite long with the test module in them. Adding the environment name will make them too long.